src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -325,6 +315,58 @@ private static string GetUri(HttpRequest request)
             return builder.ToString();
         }
 
+        private static string GetRoute(IDictionary<string, object> routeValues)
+        {
+            string route = null;
+
+            if (routeValues.Count > 0)
+            {
+                object controller;
+                routeValues.TryGetValue("controller", out controller);
+                string controllerName = (controller == null) ? string.Empty : controller.ToString();
+
+                if (!string.IsNullOrEmpty(controllerName))
+                {
+                    route = controllerName;
+
+                    if (routeValues.TryGetValue("action", out var action) && action != null)
+                    {
+                        route += "/" + action.ToString();
+                    }
+
+                    if (routeValues.Keys.Count > 2)
+                    {
+                        // Add parameters
+                        var sortedKeys = routeValues.Keys
+                            .Where(key =>
+                                !string.Equals(key, "controller", StringComparison.OrdinalIgnoreCase) &&
+                                !string.Equals(key, "action", StringComparison.OrdinalIgnoreCase) &&
+                                !string.Equals(key, "!__route_group", StringComparison.OrdinalIgnoreCase))
+                            .OrderBy(key => key, StringComparer.OrdinalIgnoreCase)
+                            .ToArray();
```

Comment:
lots of new allocs with this change. it may not better to put them under optional flag.

Reply:
We could also call the `Enrich` option here: `this.options.Enrich?.Invoke(activity, "OnCustomActivity", payload);` if we don't want all the users to incur this additional cost.